### PR TITLE
fix:   maximum hook update

### PR DESCRIPTION
**Pull request by mortada-codes**
This PR...

## Changes

-

## Fixes

-

## How to test it

-

## Screenshots

-

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

**Comment by devcatalin** [comment]
I've made some changes thinking that this might fix the FormEditor but it looks like the Form is still infintely re-rendering for some reason. We'll have to test it more.

**Comment by mortada-codes** [comment]
> I've made some changes thinking that this might fix the FormEditor but it looks like the Form is still infintely re-rendering for some reason. We'll have to test it more.

working on it

File: src/components/molecules/FormEditor/FormEditor.tsx [hunk 1]
```diff
@@ -92,30 +82,7 @@ const FormEditor: React.FC<IProps> = props => {
...
-      const fileContent = fs.readFileSync(absoluteFilePath, 'utf8');
-      const content = mergeManifests(fileContent, formString);
-      return content.trim() !== fileContent.trim();
-    }
-
-    return false;
-  }, [fileMap, formData, selectedFilePath, selectedResource?.text]);
-
-  useWhatChanged([fileMap, selectedFilePath, selectedResource, formData, changed]);
+  useWhatChanged([fileMap, selectedFilePath, selectedResource, formData]);
```
Comment (devcatalin) [inline src/components/molecules/FormEditor/FormEditor.tsx:85]: Let's remove this now since it was only used for debugging!

**Review by devcatalin — CHANGES_REQUESTED** [review]

File: src/redux/thunks/saveResourceChanges.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+import {setAutosavingError, setAutosavingStatus} from '@redux/reducers/main';
+import {selectedFilePathSelector} from '@redux/selectors';
+import {getAbsoluteFilePath} from '@redux/services/fileEntry';
+import {mergeManifests} from '@redux/services/manifest-utils';
+
+import {AppDispatch} from '@shared/models/appDispatch';
+import {RootState} from '@shared/models/rootState';
+
+export const saveResourceChanges = createAsyncThunk<void, string, {dispatch: AppDispatch; state: RootState}>(
```
Comment (devcatalin) [inline src/redux/thunks/saveResourceChanges.ts:14]: Let's rename this to `saveFormEditorResource` so we don't confuse it with something else

File: src/redux/thunks/readResourceFile.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+import fs from 'fs/promises';
+import log from 'loglevel';
+
+import {getAbsoluteFilePath} from '@redux/services/fileEntry';
+
+import {AppDispatch} from '@shared/models/appDispatch';
+import {RootState} from '@shared/models/rootState';
+
+export const readResourceFile = createAsyncThunk<string | undefined, string, {dispatch: AppDispatch; state: RootState}>(
```
Comment (devcatalin) [inline src/redux/thunks/readResourceFile.ts:11]: We can rename this to `readFileThunk` since it only reads a file

**Review by devcatalin — CHANGES_REQUESTED** [review]

**Review by devcatalin — APPROVED** [review]

**Pull request merged**